soc/adi/max32/soc.yml

```diff
@@ -7,6 +7,7 @@ family:
   - name: max32655
     cpuclusters:
     - name: m4
+  - name: max32666
```

Comment:
Since this is a dual-core part, let's define the first core as a cpucluster so it doesn't have to get renamed in the future when we add support for the second core.

Reply:
Sure done,
Related board files and test files updated.